ams/server/src/main/java/com/netease/arctic/server/optimizing/maintainer/IcebergTableMaintainer.java

```diff
@@ -203,11 +203,19 @@ private void expireSnapshots(long olderThan, Set<String> exclude) {
             })
         .cleanExpiredFiles(true)
         .commit();
-    if (arcticFileIO().supportFileSystemOperations()) {
-      parentDirectory.forEach(
-          parent -> TableFileUtil.deleteEmptyDirectory(arcticFileIO(), parent, exclude));
+    try (ArcticFileIO io = arcticFileIO()) {
+      parentDirectory.forEach(parent -> TableFileUtil.deleteEmptyDirectory(io, parent, exclude));
+    } catch (IllegalStateException e) {
+      // Doesn't support directory operations
+      throw e;
```

Comment:
I think we should support FileIO which do not support directory operations and skip the deleting empty directories operations as as we did before rather than throwing an exception.